src/test/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommandTest.java

```diff
@@ -20,41 +22,72 @@
 import seedu.sellsavvy.model.UserPrefs;
 import seedu.sellsavvy.model.order.Order;
 import seedu.sellsavvy.model.order.OrderList;
+import seedu.sellsavvy.model.order.Status;
+import seedu.sellsavvy.model.order.StatusEqualsKeywordPredicate;
 import seedu.sellsavvy.model.person.Person;
+import seedu.sellsavvy.testutil.OrderBuilder;
 
 public class MarkOrderCommandTest {
 
     private Model model;
+    private Person personToMarkOrderUnder;
 
     @BeforeEach
     public void setUp() {
         model = new ModelManager(getTypicalAddressBook(), new UserPrefs()).createCopy();
-        Person personToMarkOrderUnder = model.getFilteredPersonList().get(INDEX_FOURTH_PERSON.getZeroBased());
+        personToMarkOrderUnder = model.getFilteredPersonList().get(INDEX_FOURTH_PERSON.getZeroBased());
         model.updateSelectedPerson(personToMarkOrderUnder);
     }
 
     @Test
-    public void execute_validIndexOrderList_success() {
+    public void execute_validIndexUnfilteredOrderList_success() {
         MarkOrderCommand markOrderCommand = new MarkOrderCommand(INDEX_FIRST_ORDER);
 
         Model expectedModel = model.createCopy();
-        Order orderToBeMarked = expectedModel.getFilteredPersonList().get(INDEX_FOURTH_PERSON.getZeroBased())
-                .getOrderList().get(INDEX_FIRST_ORDER.getZeroBased());
-        Order markedOrder = MarkOrderCommand.createMarkedOrder(orderToBeMarked);
+        Order orderToBeMarkedInUnfilteredList = getOrderByIndex(expectedModel, INDEX_FIRST_ORDER);
+        OrderBuilder orderBuilder = new OrderBuilder(orderToBeMarkedInUnfilteredList);
+        Order markedOrder = orderBuilder.withStatus(Status.COMPLETED).build();
```

Comment:
Order builder can be used as new OrderBuilder().withStatus() iirc, for reference see typicalOrders